### PR TITLE
Add new logo and diagram to Sphinx/RTD docs

### PR DESCRIPTION
**Pull request by LauWien**
This PR adds the new `message-ix` logo and diagram to the documentation and is the corresponding PR for https://github.com/iiasa/message_data/pull/368 opened by @khaeru.

## How to review

- Read the diff and note that the CI checks all pass.
- Run a specific code snippet or command and check the output.
- View the RTD docs preview at **https://docs.messageix.org/en/feature-new_logo/** look at:
  - the combined IIASA-`message-ix` logo top left,
  - the favicon, and
  - the new diagram under "Getting started".

## PR checklist

- [x] Continuous integration checks all ✅
- [x] Update release notes.

### CODE REVIEW

File: doc/index.rst [hunk 1]
```diff
@@ -49,6 +43,12 @@ Then, continue with the:
    tutorials
    Publications, Projects, and Tools <usage>
 
+.. figure:: _static/ix_features.svg
+   :width: 360px
+   :align: center
+
+   Features of ``ixmp``, ``message_ix``, and related packages(:cite:`huppmann_messageix_2018`)
```
Comment (khaeru) [inline doc/index.rst:50]: Space before left parenthesis, and period afterwards.

**Review by khaeru — APPROVED** [review]
Looks great, but let's hold off on merging pending any comments during the weekly meeting tomorrow.

File: doc/index.rst (hunk 1 above)
Comment (khaeru) [reply doc/index.rst:50]: Actually, if you look at the built docs, the citation style is numeric, so this comes out as “packages([3])”. I think the parentheses can be removed altogether.

**Comment by LauWien** [comment]
Thank you for your review!

**Comment by khaeru** [comment]
@LauWien since we have no comments on this since the morning meeting, IMO it is okay to rebase and merge.

**Comment by LauWien** [comment]
Maybe a general question: as we merged other PRs before into `main` the bullet point for this PR in the release notes is not on top of the list anymore. Is that fine or is it better to move it up again, but in one way changing a bit the history?

**Pull request merged**